src/lib/falcor/FalcorRouter.js

```diff
@@ -914,6 +915,106 @@ export default class FalcorRouter extends BaseRouter.createClass([
     },
   },
   {
+    route: "search['teams'][{keys:queries}][{integers:indices}]",
+    get: (pathSet) => {
+      return new Promise((resolve) => {
+        let minIndex = pathSet.indices[0];
+        let maxIndex = pathSet.indices[0];
+        pathSet.indices.forEach((index) => {
+          if (index < minIndex) {
+            minIndex = index;
+          }
+          if (index > maxIndex) {
+            maxIndex = index;
+          }
+        });
+        db.searchTeamsQuery(pathSet.queries, minIndex, maxIndex).then((data) => {
+          // Map all the indices down to fit the indices returned by the db call
+          pathSet.indices = pathSet.indices.map((index) => {return index-minIndex});
+          const results = [];
+          _.forEach(data, (queryResults, query) => {
+            pathSet.indices.forEach((index) => {
+              if (index < queryResults.length) {
+                results.push({
+                  path: ['search', 'teams', query, index],
+                  value: $ref(['teamsBySlug', queryResults[index]]),
+                });
+              }
+            });
+          });
+          resolve(results);
+        });
+      });
+    },
+  },
+  {
+    route: "teamsBySlug[{keys:slugs}]['id', 'name', 'description']",
+    get: (pathSet) => {
+      return new Promise((resolve) => {
+        const requestedFields = pathSet[2];
+        db.teamQuery(pathSet.slugs, requestedFields).then((data) => {
+          const results = [];
+          data.forEach((team) => {
+            requestedFields.forEach((field) => {
+              results.push({
+                path: ["teamsBySlug", team.slug, field],
+                value: team[field],
+              });
+            });
+          });
+          resolve(results);
+        });
+      });
+    },
+    set: (jsonGraphArg) => {
+      return new Promise((resolve, reject) => {
+        const teamsBySlug = jsonGraphArg.teamsBySlug;
+        db.updateTeams(teamsBySlug).then((flag) => {
+          if (!flag) {
+            throw new Error("For unknown reasons updateTeams returned a non-true flag");
+          }
+          const results = [];
+          _.forEach(teamsBySlug, (teamObject, slug) => {
+            _.forEach(teamObject, (value, field) => {
+              results.push({
+                path: ['teamsBySlug', slug, field],
+                value,
```

Comment:
@emilgoldsmith check [ES6 syntax](https://ariya.io/2013/02/es6-and-object-literal-property-value-shorthand) if you thought this was the syntax error, but I will adhere to your style and use `value: value` anyway.